File: python/game.py
import random

guess = 0
attempt = 10
game = True
random_number = random.randint(1, 100)

while game:
    guess = int(input("Type a number to 1 to 100 -->"))

    if guess == random_number:
        print("You win")
        game = False
    elif guess > random_number:
        print("the number is smaller")
    else:
        print("the number is greater")

        attempt -= 1
        if attempt == 0:
            print(f"you lose the number is: {random_number}")
            game = False
        else:
            print(f"you have {attempt} chances")

File: java/g/src/game.java
import java.util.Random;
import java.util.Scanner;

public class game {

    public static void main(String[] args) {
        var guess = 0;
        var attempt = 10;
        boolean game = true;
        Scanner typeNumber = new Scanner(System.in);
        Random random = new Random();
        int randomNumber = random.nextInt(100) + 1;


        while (game) {
            System.out.print("Type a number to 1 to 100 --> ");
            String number = typeNumber.next();
            guess = Integer.parseInt(number);

            if (guess == randomNumber) {
                System.out.println("you win");
                game = false;
            } else if (guess > randomNumber) {
                System.out.println("the number is smaller");
            } else {
                System.out.println("the number is greater");
            }

            attempt--;
            if (attempt == 0) {
                System.out.printf("you lose, the number is %s", randomNumber);
                game = false;
            } else {
                System.out.printf("You have %s:\n", attempt, "chances");
            }
        }
        typeNumber.close();
    }
}
